platforms/software/dependency-management/src/main/java/org/gradle/api/internal/attributes/matching/DefaultAttributeMatcher.java

```diff
@@ -43,6 +42,7 @@
 public class DefaultAttributeMatcher implements AttributeMatcher {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(DefaultAttributeMatcher.class);
+    private static final AttributeMatchingExplanationBuilder EXPLANATION_BUILDER = AttributeMatchingExplanationBuilder.logging();
```

Comment:
I'm not sure this is safe to do, the instance changes based on the logging level available, which may not be static over the lifetime of a daemon? I'm not sure because the ways in which it can change are unclear, but I would be wary of making this optimization without figuring that out, and it costs very little to not store this.

Reply:
I can inline this